backend/src/models/invoice_resources.js

```diff
@@ -0,0 +1,96 @@
+const dbConnection = require("../database/databaseConnection");
+const { knex, dataBaseSchemas } = dbConnection();
+
+const table = `${dataBaseSchemas().data}.invoice_detail`;
+const contractResourceTable = `${dataBaseSchemas().data}.contract_resource`;
+const resourceTable = `${dataBaseSchemas().data}.resource`;
+const fiscalTable = `${dataBaseSchemas().data}.fiscal_year`;
+
+// Get all resources by invoice id.
+const findAllByInvoiceId = (invoiceId) => {
+  return knex
+    .select(
+      "*",
+      { resource_assignment: knex.raw("r.resource_last_name || ', ' || r.resource_first_name") },
+      { rate: knex.raw("i.rate::numeric::float8") },
+      { amount: knex.raw("(i.rate * i.unit_amount)::numeric::float8") },
+      { hours: "i.unit_amount" },
+      "i.id"
+    )
+    .from(`${table} as i`)
+    .join(`${contractResourceTable} as cr`, { "i.contract_resource_id": "cr.id" })
+    .leftJoin(`${resourceTable} as r`, { "cr.resource_id": "r.resource_id" })
+    .where("i.invoice_id", invoiceId);
+};
+
+// Get one.
+const findById = (id) => {
+  return knex
+    .select(
+      "i.id",
```

Comment:
we should be careful of abbreviation of tables to single letters unless it is the only table, as this one at min should be id (invoice_detail).  Just makes reading more difficult and easier to make mistakes, for what we are saving here, sometimes it might even be better to have the fully qualified table name.